integration-tests/rpc-get-protocol-constants.spec.ts

```diff
@@ -0,0 +1,528 @@
+import { TezosToolkit } from "@taquito/taquito";
+import { ConstantsResponse } from "@taquito/rpc";
+
+const rpcUrl = 'https://api.tez.ie/rpc/mainnet';
+const delphinetUrl = 'https://api.tez.ie/rpc/delphinet';
+const Tezos = new TezosToolkit(rpcUrl);
+const TezosDelphi = new TezosToolkit(delphinetUrl);
+
+describe('Fetch constants for all protocols on Mainnet', () => {
+    // tests
+    it('succesfully fails at fetching constants for level 0', async (done) => {
+        try {
+            await Tezos.rpc.getConstants({ block: "0" });
+            expect.assertions(1);
+        } catch (ex) {
+            expect(ex.message).toMatch('Http error response: (404) ');
+        }
+        done();
+    })
+
+    // TODO: Test that we aren't getting any extra properties. Currently, tests will pass even if we had extra properties in JSON response
+    it('successfully fetches Proto1 constants at level 1', async (done) => {
+        try {
+            // Get constants for protocol
+            const constants = await Tezos.rpc.getConstants({ block: "1" });
+            // expect(constants).toBeInstanceOf(ConstantsResponse);
+            expect(constants).toHaveProperty("proof_of_work_nonce_size", 8);
+            expect(constants).toHaveProperty("nonce_length", 32);
+            expect(constants).toHaveProperty("max_revelations_per_block", 32);
+            expect(constants).toHaveProperty("max_operation_data_length", 16384);
+            expect(constants).toHaveProperty("preserved_cycles", 5);
+            expect(constants).toHaveProperty("blocks_per_cycle", 4096);
+            expect(constants).toHaveProperty("blocks_per_commitment", 32);
+            expect(constants).toHaveProperty("blocks_per_roll_snapshot", 256);
+            expect(constants).toHaveProperty("blocks_per_voting_period", 32768);
+            expect(constants.time_between_blocks.toString()).toEqual("60,75");
+            expect(constants).toHaveProperty("endorsers_per_block", 32);
+            expect(constants.hard_gas_limit_per_operation.toString()).toEqual("400000");
+            expect(constants.hard_gas_limit_per_block.toString()).toEqual("4000000");
+            expect(constants.proof_of_work_threshold.toString()).toEqual("70368744177663");
+            expect(constants.tokens_per_roll.toString()).toEqual("10000000000");
+            expect(constants).toHaveProperty("michelson_maximum_type_size", 1000);
+            expect(constants.seed_nonce_revelation_tip.toString()).toEqual("125000");
+            expect(constants.origination_burn?.toString()).toEqual("257000");
+            expect(constants.block_security_deposit.toString()).toEqual("0");
+            expect(constants.endorsement_security_deposit.toString()).toEqual("0");
+            if (constants.block_reward) {
+                expect(constants.block_reward?.toString()).toEqual("0");
+            } else {
+                expect(constants.block_reward).toBeUndefined();
+            }
```

Comment:
`block_reward` is defined in proto1, so I don't think will should expect it to be undefined. I suggest removing the else part :
```
else {
    expect(constants.block_reward).toBeUndefined();
}
```
(same for the other constants in the other tests when they are supposed to be defined)

Reply:
removed all cases where else {
                expect(constants.block_reward).toBeUndefined();

Reply:
removed:
else {
    expect(constants.block_reward).toBeUndefined();
}